File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1

        while l < r:
            mid = l + (r - l) // 2

            # If the middle element is smaller than the next element, the peak must be to the right
            if nums[mid] < nums[mid + 1]:
                l = mid + 1
            else:  # Otherwise, the peak must be to the left or at mid
                r = mid

        return l

File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int n = nums.length;
        boolean[] ans = new boolean[n];
        Arrays.fill(ans, false);

        for (int i = 0; i < n; i++) {
            if (i == 0 && n>1 && nums[i] > nums[i + 1])
                ans[i] = true;
            if (i == n - 1 && n>1 && nums[i] > nums[i - 1])
                ans[i] = true;
            if (i > 0 && i < n - 1 && nums[i] > nums[i + 1] && nums[i] > nums[i - 1]) {
                ans[i] = true;
            }
        }
        for (int i = 0; i < n; i++) {
            if (ans[i])
                return i;
        }
        return (n == 1) ? 0 : -1;
    }
}